Update sensor lpcmp compatible name

ZhaoxiangJin opened the pull request:
1. Removed `nxp,sensor-lpcmp.yaml` and merged its properties into `nxp,lpcmp.yaml`. Now a single dt-binding (compatible: `nxp,lpcmp`) is shared by two drivers: the sensor API-based LPCMP driver and the comparator API-based LPCMP driver.
2. Updated the driver and samples accordingly. This change reverts the dt compatible modification introduced by https://github.com/zephyrproject-rtos/zephyr/pull/100998
3. Simplified the LPCMP driver migration notes by removing outdated information about the devicetree compatible renaming. The compatible `nxp,lpcmp` is now shared between both the sensor API-based and comparator API-based LPCMP drivers, so the previous note about renaming to `nxp,sensor-lpcmp` is no longer accurate.

henrikbrixandersen reviewed (changes requested):
As already discussed on Discord, a compatible should not include the name of the SDK/HAL used for implementing the driver.

henrikbrixandersen commented on dts/bindings/sensor/nxp,mcx-lpcmp.yaml:
```diff
@@ -3,7 +3,7 @@
 description: NXP low-power analog comparator (LPCMP)
 
-compatible: "nxp,sensor-lpcmp"
+compatible: "nxp,mcx-lpcmp"
```
This device is not - and never was - limited to the MCX families.

butok commented:
> As already discussed on Discord, a compatible should not include the name of the SDK/HAL used for implementing the driver.

Hi @henrikbrixandersen 
I understand your point, but this is already established practice:
- Zephyr NXP Shim drivers based on MCUXpresso drivers (NXP HAL) contain "mcux" in their name and binding.
- Zephyr NXP Native drivers (not shims) do not use "mcux".

butok commented:
> > Hi @henrikbrixandersen I understand your point, but this is already established practice:
> > 
> > * Zephyr NXP Shim drivers based on MCUXpresso drivers (NXP HAL) contain "mcux" in their name and binding.
> > * Zephyr NXP Native drivers (not shims) do not use "mcux".
> 
> That may well be, but that practice is wrong. Devicetree compatibles describe the hardwares programming _model_, not the driver, not the SDK used for implementing that driver.

I agree with your point. 
But if this is so terrible, why has nobody stopped or noticed this in all these ten years?

henrikbrixandersen commented:
> I agree with your point.

Super.

> But if this is so terrible, why has nobody stopped or noticed this in all these ten years?

I don't know that, but to be fair we do not have that many DTS bindings with "mcux" in their compatible string:

```
$ git grep -E 'compatible:.*mcux.*' dts/bindings/|wc -l
10
```

We do, however, have a lot of driver implementation files with "mcux" in their name, denoting that it's a shim driver on top of the MCUXpresso SDK (which is fine):

```
$ git ls-files 'drivers/***/*mcux*.c'|wc -l
127
```

JarmouniA commented:
> But if this is so terrible, why has nobody stopped or noticed this in all these ten years?

Honestly, I always thought it was a SoC family name (confusion with MCX), similar to STM32, nRF...

ZhaoxiangJin committed:
dts: drivers: remove nxp,sensor-lpcmp.yaml

1. Removed nxp,sensor-lpcmp.yaml and merged its
properties into nxp,lpcmp.yaml. Now a single
dt-binding (compatible: nxp,lpcmp) is shared
by two drivers: the sensor API-based LPCMP
driver and the comparator API-based LPCMP driver.

2. Updated the driver and samples accordingly.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
drivers: sensor: mcux_lpcmp: Removed redundant help information

Removed redundant help information

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
doc: releases: update LPCMP deprecation notes for 4.4

Simplified the LPCMP driver migration notes by removing outdated
information about the devicetree compatible renaming. The compatible
nxp,lpcmp is now shared between both the sensor API-based and
comparator API-based LPCMP drivers, so the previous note about
renaming to nxp,sensor-lpcmp is no longer accurate.

Moved the comparator API-based LPCMP driver information to the
'New Drivers' directory.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin commented on drivers/sensor/nxp/mcux_lpcmp/mcux_lpcmp.c:
```diff
@@ -5,7 +5,7 @@
  */
 
-#define DT_DRV_COMPAT nxp_sensor_lpcmp
+#define DT_DRV_COMPAT nxp_mcx_lpcmp
```
single dt-binding for different API based driver is feasible in this case, updated in the PR, please review.

JarmouniA commented on dts/bindings/comparator/nxp,lpcmp.yaml:
```diff
@@ -120,3 +120,56 @@ properties:
+  enable-output-invert:
+    type: boolean
+    description: |
+      Decide whether to invert the comparator output.
```
```suggestion
      Invert the comparator output.
```

kartben commented on doc/releases/release-notes-4.4.rst:
```diff
@@ -93,10 +93,7 @@ Deprecated APIs and options
-    * The original :dtcompatible:`nxp,lpcmp` has been renamed to :dtcompatible:`nxp,sensor-lpcmp`, applications
-      using the old compatible must update their devicetree nodes to use the ``mcux_lpcmp`` driver. The
-      :dtcompatible:`nxp,lpcmp` now is used for the ``comparator_nxp_lpcmp`` driver. (:github:`100998`).
+      ``gpio_loopback`` test case (:zephyr_file:`tests/drivers/comparator/gpio_loopback/src/test.c`). (:github:`100998`).
```
Why is this bullet point in the deprecated section? This seems wrong

JarmouniA replied:
> looks like something for the migration guide?

It is a test, the user is not really expected to take any migration action.

But also, I don't see why we have to mention it at all (in the migration guide or release notes).

JarmouniA commented on drivers/sensor/nxp/mcux_lpcmp/Kconfig:
```diff
@@ -5,19 +5,12 @@
 	bool "NXP LPCMP driver [DEPRECATED]"
-	default y
-	depends on DT_HAS_NXP_SENSOR_LPCMP_ENABLED
+	depends on DT_HAS_NXP_LPCMP_ENABLED
```
```suggestion
	depends on DT_HAS_NXP_LPCMP_ENABLED
	depends on !COMPARATOR_NXP_LPCMP 
```
To avoid having both drivers enabled at the same time.

ZhaoxiangJin replied:
added `depends on !COMPARATOR_NXP_LPCMP`.

ZhaoxiangJin commented:
> > I prefer to rebase to the latest main branch to ensure that the features in the PR pass tests on the most up-to-date codebase, avoiding any functional issues caused by time lags during the commit integration process.
> 
> CI rebases the commits on the latest main each time it runs.

that is good, but it can not check the run result, right?
